apps/lib/s_cb.c

```diff
@@ -770,13 +776,42 @@ int generate_cookie_callback(SSL *ssl, unsigned char *cookie,
     BIO_ADDR_rawaddress(peer, buffer + sizeof(port), NULL);
 
     /* Calculate HMAC of buffer using the secret */
-    HMAC(EVP_sha1(), cookie_secret, COOKIE_SECRET_LENGTH,
-         buffer, length, cookie, cookie_len);
-
+    hmac = EVP_MAC_fetch(NULL, SN_hmac, NULL);
```

Comment:
Should we really be using the `SN_*` defines for this sort of thing? They are related to the old legacy object database. Wouldn't it just be better to use "HMAC" explicitly, or provide new defines?

Reply:
Good point.  I've used the `SN_` form inside libcrypto and continued doing it here.  Since this isn't libcrypto, it makes sense to just use the string form directly.

Reply:
I wouldn't use it in libcrypto either

Reply:
All the `SN_` instanced are changed.